src/middleware.rs

```diff
@@ -58,74 +55,49 @@ impl PluginFor<Response> for HandlebarsEngine {
     }
 }
 
-fn read_file(path: &Path) -> Option<String> {
-    if let Ok(mut file) = File::open(path) {
-        let mut buf = String::new();
-        if file.read_to_string(&mut buf).is_ok() {
-            Some(buf)
-        } else {
-            info!("Failed to read file {}, skipped", path.display());
-            None
-        }
-    } else {
-        info!("Failed to open file {}, skipped.", path.display());
-        None
+
+impl HandlebarsEngine {
+    /// #[Deprecated], for backward compaitibility only
+    pub fn new(prefix: &str, suffix: &str) -> HandlebarsEngine {
+        let mut hbs = HandlebarsEngine::new2();
+        hbs.add(Box::new(DirectorySource::new(prefix, suffix)));
+        hbs.reload();
+        hbs
     }
-}
 
-fn filter_file(entry: &DirEntry, suffix: &str) -> bool {
-    entry.file_name().to_str()
-        .and_then(|s| Some(s.starts_with(".") || s.starts_with("#") || !s.ends_with(suffix)))
-        .unwrap_or(false)
-}
+    pub fn from(prefix: &str, suffix: &str, custom: Handlebars) -> HandlebarsEngine {
```

Comment:
`from` is previous API when you want the middleware to manager template directory for you and also need to register custom helpers with a custom `Handlebars`.